pennylane_lightning/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -763,6 +411,208 @@ class StateVectorKokkos final
         applyMatrix(matrix.data(), wires, inverse);
     }
 
+    void applyNamedOperation(const std::string &opName,
+                             const std::vector<size_t> &wires,
+                             bool inverse = false,
+                             const std::vector<fp_t> &params = {}) {
+        switch (gates_indices_[opName]) {
+        case GateOperation::PauliX:
+            applyGateFunctor<pauliXFunctor, 1>(wires, inverse, params);
+            return;
```

Comment:
Beautiful code reduction.
How should we expect this change from an unordered map to a switch case scenario to impact performance?

Reply:
For the most part, I'd expect it to be faster, but both should be order of 1-10us in either case. I think whatever improves readability is likely the best option.